feat(cocoapods): Add CocoaPods deprecation warnings to Podspec, README, and CHANGELOG (#16172)
firebase/firebase-ios-sdk: FirebaseCore/CHANGELOG.md

# Unreleased
- [fixed] Remove extra comma in Package.swift that caused SPM resolution
  failures for some configurations.

> - [deprecated] The Firebase Apple SDK via CocoaPods is deprecated in favor of

Reviewer: nit: `Installing the Firebase Apple SDK` instead of just `The Firebase Apple SDK`

Author: Fixed, this is better but I still don't like the "installing Firebase is deprecated" gist because installing will always be supported. Let me wordsmith some more...

> - [deprecated] The Firebase Apple SDK via CocoaPods is deprecated in favor of

Author: ```suggestion
- [deprecated] Installing the Firebase Apple SDK using CocoaPods is deprecated in favor of
```

  Swift Package Manager. Firebase products remain fully supported, but new
  versions will no longer be published to CocoaPods after **October 2026**.

>   See the [migration guide](https://firebase.google.com/docs/ios/cocoapods-deprecation) for more information.

Author: ```suggestion
- [deprecated] The Firebase Apple SDK via CocoaPods is deprecated in favor of
  Swift Package Manager. Firebase products remain fully supported, but new
  versions will no longer be published to CocoaPods after **October 2026**.
  See the [migration guide](https://firebase.google.com/docs/ios/cocoapods-deprecation) for more information.
```

# Unreleased
- [fixed] Remove extra comma in Package.swift that caused SPM resolution
  failures for some configurations.

> - todo: deprecation announcement and link to migration guide

Author: ```suggestion
- [deprecated] The Firebase Apple SDK via CocoaPods is deprecated in favor of 
  Swift Package Manager. Firebase products remain fully supported, but new
  versions will no longer be published to CocoaPods after **October 2026**.
  See the [migration guide](https://firebase.google.com/docs/ios/cocoapods-deprecation) for more information.
```
